### PR TITLE
[fix] Update to node 18 in straggling references

### PR DESCRIPTION
**Pull request by casthewiz**
Updates various configs in various places to node 18 from node 12/14. 

Backs out changes to lambdas - was observing that ECS tasks were not starting after Slack deployment of a docset. 

Adds Serverless Prune - see https://jira.mongodb.org/browse/DOP-3563. This was needed for testing purposes - we hit our code storage limit while attempting to debug above ECS issue.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,10 +28,10 @@
     "@aws-sdk/client-ssm": "^3.54.1",
     "@babel/core": "^7.4.4",
     "@babel/polyfill": "^7.4.4",
+    "@types/aws-lambda": "^8.10.92",
     "@types/config": "^0.0.39",
     "@types/tsscmp": "^1.0.0",
     "async-retry": "^1.2.3",
-    "aws-lambda": "^1.0.7",
```
Comment (seungpark) [inline package.json:34]: why are we removing aws-lambda here? is this part of node 18 upgrade? https://aws.amazon.com/blogs/compute/node-js-18-x-runtime-now-available-in-aws-lambda/

File: api/controllers/v1/jobs.ts [hunk 2]
```diff
@@ -137,35 +134,42 @@ async function stopECSTask(taskId: string, consoleLogger: ConsoleLogger) {
...
   const client = new mongodb.MongoClient(c.get('dbUrl'));
   await client.connect();
   const db = client.db(c.get('dbName'));
   const env = c.get<string>('env');
 
   const jobRepository = new JobRepository(db, c, consoleLogger);
+  // TODO: Make fullDocument be of type Job, validate existence
   const fullDocument = await jobRepository.getJobById(jobId);
-
-  if (!fullDocument) {
```
Comment (seungpark) [inline api/controllers/v1/jobs.ts:150]: should this error be preserved?

File: api/controllers/v1/jobs.ts [hunk 3]
```diff
@@ -137,35 +134,42 @@ async function stopECSTask(taskId: string, consoleLogger: ConsoleLogger) {
   await ecs.stopZombieECSTask(taskId);
 }
 
-async function NotifyBuildSummary(jobId: string): Promise<void> {
+async function retry(message: JobQueueMessage, consoleLogger: ConsoleLogger, url: string): Promise<any> {
```
Comment (seungpark) [inline api/controllers/v1/jobs.ts:137]: seems like retry function is not used?

File: api/controllers/v1/dochub.ts [hunk 4]
```diff
@@ -1,48 +1,17 @@
...
-  if (!FASTLY_DOCHUB_MAP) {
-    return {
-      statusCode: 500,
-      body: 'Fastly dochub map is not defined',
-    };
-  }
-
-  const creds = new CDNCreds(FASTLY_DOCHUB_SERVICE_ID, FASTLY_DOCHUB_TOKEN);
-  await new FastlyConnector(new ConsoleLogger()).upsertEdgeDictionaryItem(pair, FASTLY_DOCHUB_MAP, creds);
+  // TODO: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
```
Comment (seungpark) [inline api/controllers/v1/dochub.ts:11]: minor. can we use non-null assertion operator (!) after env varialbes

File: api/controllers/v1/github.ts [hunk 5]
```diff
@@ -1,35 +1,22 @@
...
 // This function will validate your payload from GitHub
 // See docs at https://developer.github.com/webhooks/securing/#validating-payloads-from-github
-function validateJsonWebhook(request: APIGatewayEvent, secret: string) {
-  const expectedSignature =
-    'sha256=' +
-    crypto
-      .createHmac('sha256', secret)
-      .update(request.body ?? '')
-      .digest('hex');
+function validateJsonWebhook(request: any, secret: string) {
```
Comment (seungpark) [inline api/controllers/v1/github.ts:10]: minor. add TODO for using @types/aws-lambda

File: api/controllers/v1/github.ts [hunk 6]
```diff
@@ -40,7 +27,7 @@ async function prepGithubPushPayload(
     title: githubEvent.repository.full_name,
     user: githubEvent.pusher.name,
     email: githubEvent.pusher.email,
-    status: JobStatus.inQueue,
+    status: 'inQueue',
```
Comment (seungpark) [inline api/controllers/v1/github.ts:30]: why did we remove JobStatus here

File: api/controllers/v1/jobs.ts [hunk 7]
```diff
@@ -11,22 +11,13 @@ import { Job, JobStatus } from '../../../src/entities/job';
...
 
-export const TriggerLocalBuild = async (event: APIGatewayEvent): Promise<APIGatewayProxyResult> => {
+export const TriggerLocalBuild = async (event: any = {}, context: any = {}): Promise<any> => {
+  const client = new mongodb.MongoClient(c.get('dbUrl'));
+  await client.connect();
+  const db = client.db(c.get('dbName'));
   const consoleLogger = new ConsoleLogger();
   const sqs = new SQSConnector(consoleLogger, c);
-
-  if (!event.body) {
```
Comment (seungpark) [inline api/controllers/v1/jobs.ts:20]: removed error here?

File: api/controllers/v1/jobs.ts [hunk 8]
```diff
@@ -47,23 +38,26 @@ export const TriggerLocalBuild = async (event: APIGatewayEvent): Promise<APIGate
...
    * if it is completed call NotifyBuildSummary
    */
-  const messages = event.Records;
+  const messages: JobQueueMessage[] = event.Records;
   await Promise.all(
-    messages.map(async (message: SQSRecord) => {
+    messages.map(async (message: any) => {
       const consoleLogger = new ConsoleLogger();
       const body = JSON.parse(message.body);
+      let queueUrl = '';
```
Comment (seungpark) [inline api/controllers/v1/jobs.ts:54]: is this var `queueUrl` ever used

File: api/controllers/v1/jobs.ts [hunk 9]
```diff
@@ -47,23 +38,26 @@ export const TriggerLocalBuild = async (event: APIGatewayEvent): Promise<APIGate
...
+// TODO: use @types/aws-lambda
+export const HandleJobs = async (event: any = {}): Promise<any> => {
   /**
    * Check the status of the incoming jobs
    * if it is inqueue start a task
    * if it is inprogress call NotifyBuildProgress
    * if it is completed call NotifyBuildSummary
    */
-  const messages = event.Records;
+  const messages: JobQueueMessage[] = event.Records;
```
Comment (seungpark) [inline api/controllers/v1/jobs.ts:49]: is this correct? seeing message.body in line 53 and JobQueueMessage has no member body

File: api/controllers/v1/jobs.ts [hunk 10]
```diff
@@ -177,6 +181,9 @@ async function NotifyBuildSummary(jobId: string): Promise<void> {
     ),
     entitlement['slack_user_id']
   );
+  return {
```
Comment (seungpark) [inline api/controllers/v1/jobs.ts:184]: why are we returning status codes from these now? is that required? adds typing errors below (ln 260) from Promise<void> to Promise<any>

**Review by seungpark — CHANGES_REQUESTED** [review]
not sure why some of these changes are required, ie. returning status codes, unused vars. PTAL at comments!

File: api/controllers/v1/dochub.ts (hunk 4 above)
Comment (branberry) [reply api/controllers/v1/dochub.ts:11]: Hmm, my previous PR addressed the undefined env variables [here](https://github.com/mongodb/docs-worker-pool/blob/9991af6a332db69394e919bd4d18271b2ef1bfb1/api/controllers/v1/dochub.ts#L23). This PR might be out of date

**Comment by casthewiz** [comment]
@seungpark Good concerns - but said changes come from this [commit](https://github.com/mongodb/docs-worker-pool/pull/815/commits/e2fb237328bca70433950d90e5072edbe943dbf8) as a reversion of this [commit](https://github.com/mongodb/docs-worker-pool/commit/08bb9239492d9da1439e34a6ae7a427f44d60b72). 

ECS instances weren't starting in `preprd` with the changes in the commit present - in order to have a viable release candidate, I've reverted, and we'll look at re-adding it in with future changes.

**Comment by seungpark** [comment]
> @seungpark Good concerns - but said changes come from this [commit](https://github.com/mongodb/docs-worker-pool/pull/815/commits/e2fb237328bca70433950d90e5072edbe943dbf8) as a reversion of this [commit](https://github.com/mongodb/docs-worker-pool/commit/08bb9239492d9da1439e34a6ae7a427f44d60b72).
> 
> ECS instances weren't starting in `preprd` with the changes in the commit present - in order to have a viable release candidate, I've reverted, and we'll look at re-adding it in with future changes.

resolved the comments above that come from the <revert merge> commit. 
looking at [these commits after](https://github.com/mongodb/docs-worker-pool/pull/815/files/e2fb237328bca70433950d90e5072edbe943dbf8..c65316ca4eef23b9c1132c129af2b1ed05c5254b) and [these commits before](https://github.com/mongodb/docs-worker-pool/pull/815/commits/aa83ff0f235af193bea7d06743f4540c6dd5cd8f) the reversion, LGTM

**Pull request merged**